app/src/main/java/com/yoshitoke/weatheringwithyou/mvp/presenter/MainPresenter.kt

```diff
@@ -0,0 +1,69 @@
+package com.yoshitoke.weatheringwithyou.mvp.presenter
+
+import android.content.Context
+import android.util.Log
+import com.google.gson.Gson
+import com.yoshitoke.weatheringwithyou.mvp.Contract
+import com.yoshitoke.weatheringwithyou.mvp.model.DataClass.City
+import com.yoshitoke.weatheringwithyou.mvp.model.DataClass.WeatherInfo
+import com.yoshitoke.weatheringwithyou.utils.AssetsLoader
+import com.yoshitoke.weatheringwithyou.utils.network.RetrofitClient
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.disposables.Disposable
+import io.reactivex.schedulers.Schedulers
+import org.json.JSONException
+
+class MainPresenter(
+        private var mainView: Contract.View?,
+        private val context: Context) : Contract.Presenter {
+
+    companion object {
+        private var disposable: Disposable? = null
+        private var cityList: List<City> = listOf()
+    }
+
+    override fun loadCityList() {
```

Comment:
sort func: override => public => private

Reply:
can you explain more pls

Reply:
arrange all override function to top, below override function is public function, and the other is private function